experiment/template/eVOLVER.py

```diff
@@ -0,0 +1,629 @@
+                    while exp_overwrite not in ['y', 'n']:
+                        exp_overwrite = input('Directory aleady exists. '
+                                            'Overwrite with new experiment? (y/n): ')
+                logger.info('data directory already exists')
+                if exp_overwrite == 'y':
+                    logger.info('deleting existing data directory')
+                    shutil.rmtree(EXP_DIR)
+                else:
+                    print('Change experiment name in custom_script.py '
+                        'and then restart...')
+                    logger.warning('not deleting existing data directory, exiting')
+                    sys.exit(1)
+
+            start_time = time.time()
+
+            self.request_od_calibration()
+            self.request_temp_calibration()
+
+            logger.debug('creating data directories')
+            os.makedirs(os.path.join(EXP_DIR, 'OD'))
+            os.makedirs(os.path.join(EXP_DIR, 'temp'))
+            os.makedirs(os.path.join(EXP_DIR, 'temp_config'))
+            if OPERATION_MODE == 'turbidostat':
+                os.makedirs(os.path.join(EXP_DIR, 'pump_log'))
+                os.makedirs(os.path.join(EXP_DIR, 'ODset'))
+                os.makedirs(os.path.join(EXP_DIR, 'growthrate'))
+            elif OPERATION_MODE == 'chemostat':
+                os.makedirs(os.path.join(EXP_DIR, 'chemo_config'))
+            else:
+                os.makedirs(os.path.join(EXP_DIR, 'pump_log'))
+                os.makedirs(os.path.join(EXP_DIR, 'ODset'))
+                os.makedirs(os.path.join(EXP_DIR, 'growthrate'))
+                os.makedirs(os.path.join(EXP_DIR, 'chemo_config'))
+
+            for x in vials:
+                # make OD file
+                file_name =  "vial{0}_OD.txt".format(x)
+                file_path = os.path.join(EXP_DIR, 'OD', file_name)
+                text_file = open(file_path, "w")
+                text_file.write("Experiment: {0} vial {1}, {2}\n".format(
+                    EXP_NAME, x, time.strftime("%c")))
+                text_file.close()
+
+                # make temperature data file
+                file_name = "vial{0}_temp.txt".format(x)
+                file_path = os.path.join(EXP_DIR, 'temp', file_name)
+                text_file = open(file_path, "w").close()
+
+                # make temperature configuration file
+                file_name =  "vial{0}_tempconfig.txt".format(x)
+                file_path = os.path.join(EXP_DIR, 'temp_config', file_name)
+                text_file = open(file_path, "w")
+                text_file.write("Experiment: {0} vial {1}, {2}\n".format(
+                    EXP_NAME, x, time.strftime("%c")))
+                #initialize based on custom_script.choose_setup()
+                text_file.write("0,{0}\n".format(TEMP_INITIAL[x]))
+                text_file.close()
+
+                if OPERATION_MODE == 'turbidostat':
+                    # make pump log file
```

Comment:
Can you make a function do this directory creation stuff? Lots of duplicated code.

Reply:
Absolutely!